app/controllers/api/v1/patients_controller.rb

```diff
@@ -27,8 +42,24 @@ def patients_params
         :created_at,
         :updated_at,
         phone_numbers: [permitted_phone_number_params],
-        address: permitted_address_params
+        address:       permitted_address_params
       )
     end
   end
-end
+
+  def latest_record_timestamp
+    if params[:first_time].present? && params[:first_time] == 'true'
+      Time.new(0)
+    else
+      params.require(:latest_record_timestamp).to_time
+    end
+  end
+
+  def number_of_records
+    if params[:number_of_records].present?
```

Comment:
I'd prefer to rename this param to `:limit` for simplicity

Reply:
Hrm. `limit` _is_ simpler.